iml-services/iml-stats/src/main.rs

```diff
@@ -531,51 +532,93 @@ async fn main() -> Result<(), ImlStatsError> {
     Ok(())
 }
 
-#[derive(serde::Serialize, serde::Deserialize)]
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
 struct MgsFsTime {
     time: u64,
+    mgs_fs: String,
+    mgs_fs_count: Option<u32>,
 }
 
 async fn delete_existing_mgs_fs_records(
-    fs_names: String,
+    fs_names: HashSet<String>,
     client: &Client,
 ) -> Result<(), ImlStatsError> {
     let xs: Vec<MgsFsTime> = client
         .query_into(
             format!(
                 r#"
-            SELECT mgs_fs,is_mgs_fs
-            FROM target
-            WHERE mgs_fs='{}' AND is_mgs_fs=true
-            ORDER BY time ASC
-            LIMIT 2"#,
-                fs_names
+                SELECT mgs_fs, mgs_fs_count FROM 
+                    (SELECT count(is_mgs_fs) AS mgs_fs_count FROM target WHERE is_mgs_fs=true GROUP BY mgs_fs),
+                    (SELECT LAST(is_mgs_fs),time as last_time FROM target GROUP BY mgs_fs) 
+                GROUP BY mgs_fs
+                "#
             )
             .as_str(),
-            Some(Precision::Nanoseconds),
+            Some(Precision::Nanoseconds)
         )
         .await?
         .unwrap_or_default();
 
-    if xs.len() < 2 {
-        return Ok(());
-    }
+    tracing::debug!(
+        "delete existing mgs_fs records - fs_names: {:?}; query results: {:?}",
+        fs_names,
+        xs
+    );
 
-    if let Some(x) = xs.get(0) {
-        let query = format!(
-            "DELETE FROM target WHERE time < {} AND mgs_fs = '{}' AND kind = '{}'",
-            x.time,
-            fs_names,
-            TargetVariant::MGT
-        );
+    let subsets: HashMap<String, (u64, u32)> = xs
+        .into_iter()
+        .filter(|x| {
+            let mgs_fs_set = fs_names_to_hash_set(&x.mgs_fs);
+
+            let intersection = mgs_fs_set
+                .intersection(&fs_names)
+                .map(|x| x.to_string())
+                .collect::<Vec<String>>();
+
+            return !intersection.is_empty();
```

Comment:
```suggestion
            let mgs_fs_set = fs_names_to_hash_set(&x.mgs_fs);

            let intersection = mgs_fs_set.intersection(&fs_names).collect::<Vec<_>>();

            !intersection.is_empty()
```

Don't need to convert the borrow into an owned String.